src/gtfs_pathways_validation.py

```diff
@@ -0,0 +1,94 @@
+import os
+import shutil
+import logging
+import traceback
+from pathlib import Path
+from typing import Union, Any
+from .config import Settings
+
+from python_ms_core import Core
+from tdei_gtfs_csv_validator import gcv_test_release
+from tdei_gtfs_csv_validator import exceptions as gcvex
+
+ROOT_DIR = os.path.dirname(os.path.abspath(__file__))
+# Path used for download file generation.
+DOWNLOAD_FILE_PATH = f'{Path.cwd()}/downloads'
+
+logging.basicConfig()
+logger = logging.getLogger('PATHWAYS_VALIDATION')
+logger.setLevel(logging.INFO)
+
+DATA_TYPE = 'gtfs_pathways'
+SCHEMA_VERSION = 'v1.0'
+
+
+class GTFSPathwaysValidation:
+    def __init__(self, file_path=None):
+        core = Core()
+        settings = Settings()
+        self.container_name = settings.storage_container_name
+        self.storage_client = core.get_storage_client()
+        self.file_path = file_path
+        self.file_relative_path = file_path.split('/')[-1]
+        self.client = self.storage_client.get_container(container_name=self.container_name)
+
+    # Facade function to validate the file
+    # Focuses on the file name with file name validation
+    # Use `is_gtfs_pathways_valid` to do more processing
+    def validate(self) -> tuple[bool, str]:
+        return self.is_gtfs_pathways_valid()
+
+    # use this method to do the actual validation
+    # when ready to replace, replace the call in the
+    # above function.
+    def is_gtfs_pathways_valid(self) -> tuple[Union[bool, Any], Union[str, Any]]:
+        is_valid = False
+        validation_message = ''
+        root, ext = os.path.splitext(self.file_relative_path)
+        if ext and ext.lower() == '.zip':
+            downloaded_file_path = self.download_single_file(self.file_path)
+            try:
+                logger.info(f' Downloaded file path: {downloaded_file_path}')
+                gcv_test_release.test_release(DATA_TYPE, SCHEMA_VERSION, downloaded_file_path)
+                is_valid = True
+                GTFSPathwaysValidation.clean_up(downloaded_file_path)
```

Comment:
I believe this should be after the except block of the code. otherwise, in the case of invalid file, we will not clean up the local file.